cmd/azure-keyvault-controller/controller/secret_handler.go

```diff
@@ -149,9 +149,11 @@ func (h *AzureCertificateHandler) Handle() (map[string][]byte, error) {
 		return nil, fmt.Errorf("no datakey spesified for output secret")
 	}
 
+	ensureServerFirst := h.secretSpec.Spec.Output.Secret.ChainOrder == "ensureserverfirst"
+
 	log.Infof("Exporting certificate with private key: %t", exportPrivateKey)
 
-	cert, err := h.vaultService.GetCertificate(&h.secretSpec.Spec.Vault, exportPrivateKey)
+	cert, err := h.vaultService.GetCertificate(&h.secretSpec.Spec.Vault, exportPrivateKey, ensureServerFirst)
```

Comment:
When we get multiple values passed in like this, I think we should pass a option object instead (one value is OK, two is one too many). E.g:

```
type certificateOptions struct {
  exportPrivateKey bool
  ensureServerFirst bool
}
```

What do you think?

Reply:
Yes, I agree, will make the changes.

Reply:
Added a struct instead. Was unsure if I was to use factory or could use the struct directly by making it public. As I've never written anything in go before I'm a little bit unsure of the design patterns.